api/auth/index.ts

```diff
@@ -1,69 +1,130 @@
+    const tokenUrl = `${getConfigValue(SERVICES_IDAM_API_URL)}/oauth2/token`
 
-        if (!userHasAppAccess(userDetails.roles)) {
-          logger.warn('User has no application access, as they do not have a Caseworker role.')
-          doLogout(req, res)
-          return false
-        }
+    //TODO: we can move these out into proper config at some point to tidy up even further
+    const options: AuthOptions = {
+        allowRolesRegex: getConfigValue(LOGIN_ROLE_MATCHER),
+        authorizationURL: authorizationUrl,
+        callbackURL: getConfigValue(SERVICES_IDAM_OAUTH_CALLBACK_URL),
+        clientID: idamClient,
+        clientSecret: secret,
+        discoveryEndpoint: `${idamWebUrl}/o`,
+        issuerURL: issuerUrl,
+        logoutURL: idamApiPath,
+        responseTypes: ['code'],
+        scope: 'profile openid roles manage-user create-user',
+        sessionKey: 'xui-webapp',
+        tokenEndpointAuthMethod: 'client_secret_post',
+        tokenURL: tokenUrl,
+        useRoutes: true,
+    }
 
-        if (userDetails) {
-            logger.info('Setting session and cookies')
-            req.session.user = userDetails
-            res.cookie(cookieUserId, userDetails.id)
-            res.cookie(cookieToken, data.access_token)
-            res.cookie('roles', userDetails.roles.join(','))
-        }
+    const baseStoreOptions = {
+        cookie: {
+            httpOnly: true,
+            maxAge: 28800000,
+            secure: showFeature(FEATURE_SECURE_COOKIE_ENABLED),
+        },
+        name: 'xui-webapp',
+        resave: false,
+        saveUninitialized: false,
+        secret: getConfigValue(SESSION_SECRET),
     }
-    logger.info('Auth finished, redirecting')
-    req.session.save(() => {
-        res.redirect('/')
-    })
+
+    const redisStoreOptions = {
+        redisStore: {
+            ...baseStoreOptions, ...{
+                redisStoreOptions: {
+                    redisCloudUrl: getConfigValue(REDIS_CLOUD_URL),
+                    redisKeyPrefix: getConfigValue(REDIS_KEY_PREFIX),
+                    redisTtl: getConfigValue(REDIS_TTL),
+                },
+            },
+        },
+    }
+
+    const fileStoreOptions = {
+        fileStore: {
+            ...baseStoreOptions, ...{
+                fileStoreOptions: {
+                    filePath: getConfigValue(NOW) ? '/tmp/sessions' : '.sessions',
+                },
+            },
+        },
+    }
+
+    const nodeLibOptions = {
+        auth: {
+            s2s: {
+                microservice: getConfigValue(MICROSERVICE),
+                s2sEndpointUrl: `${getConfigValue(SERVICE_S2S_PATH)}/lease`,
+                s2sSecret: s2sSecret.trim(),
+            },
+        },
+        session: showFeature(FEATURE_REDIS_ENABLED) ? redisStoreOptions : fileStoreOptions,
+    }
+
+    const type = showFeature(FEATURE_OIDC_ENABLED) ? 'oidc' : 'oauth2'
+    nodeLibOptions.auth[type] = options
+    logger._logger.info('Setting XuiNodeLib options')
```

Comment:
can this logger._logger be simplified?

Reply:
This is how the 'old' logger was coded, exposing the underlying logger as _logger (I prefer to use this as it's not destructive, the old one would convert objects to [Object Object])